src/components/Formula/TopicBasedFormulas.js

```diff
@@ -83,62 +84,181 @@ const Dhis2ElementDetails = ({ dhis2Object }) => {
   );
 };
 
-const FormulaMappingDialogEditor = ({ cell }) => {
-  const formulaMapping = cell.value;
+const useStylesFormMapping = makeStyles(() => ({
+  paper: { minWidth: "500px", minHeight: "500px" },
+}));
+
+const FormulaMappingDialogEditor = props => {
+  const classes = useStylesFormMapping();
+  const cell = props.cell;
+  const formulaMapping = cell.value.formulaMapping;
+  const [newExternalReference, setNewExternalReference] = React.useState(
+    formulaMapping ? formulaMapping.externalReference : "",
+  );
+  const [dhis2Object, setDhis2Object] = React.useState(
+    formulaMapping ? dhis2LookupElement(newExternalReference) : undefined,
+  );
 
-  let dhis2Object = formulaMapping
-    ? dhis2LookupElement(formulaMapping.externalReference)
-    : undefined;
+  const [open, setOpen] = React.useState(true);
+
+  const handleChange = (_e, v) => {
+    if (v) {
+      setNewExternalReference(v.id);
+      setDhis2Object(dhis2LookupElement(v.id));
+    }
+  };
+
+  const handleConfirm = async () => {
+    if (cell.value.formulaMapping == undefined) {
+      // CREATE
+      try {
+        const resp = await externalApi()
```

Comment:
Would move this to a single method `createOrUpdate(cell_value)`, I think only the URL needs to change between the two of them.